src/CVNBot/Project.cs

```diff
@@ -183,50 +171,97 @@ void GetNamespaces(bool snamespacesAlreadySet)
+            Messages.Add("Undeletedarticle", new MessagesOption(1, "restoreRegex", false));
+            Messages.Add("Deletedarticle", new MessagesOption(1, "deleteRegex", false));
+            Messages.Add("Protectedarticle", new MessagesOption(1, "protectRegex", false));
+            Messages.Add("Unprotectedarticle", new MessagesOption(1, "unprotectRegex", false));
+            Messages.Add("Modifiedarticleprotection", new MessagesOption(1, "modifyprotectRegex", true));
+            Messages.Add("Uploadedimage", new MessagesOption(0, "uploadRegex", false));
+            Messages.Add("1movedto2",new MessagesOption(2, "moveRegex", false));
+            Messages.Add("1movedto2_redir", new MessagesOption(2, "moveredirRegex", false));
             // blockRegex is nonStrict because some wikis override the message without including $2 (block length).
             // RCReader will fall back to "24 hours" if this is the case.
             // Some newer messages (e.g. https://lmo.wikipedia.org/wiki/MediaWiki:Blocklogentry) have a third item,
             // $3 ("anononly,nocreate,autoblock"). This may conflict with $2 detection.
             // Trying (changed 2 -> 3) to see if length of time will be correctly detected using just this method:
-            GenerateRegex("MediaWiki:Blocklogentry", 3, ref blockRegex, true);
-            GenerateRegex("MediaWiki:Unblocklogentry", 0, ref unblockRegex, false);
-            GenerateRegex("MediaWiki:Reblock-logentry", 3, ref reblockRegex, false);
-            GenerateRegex("MediaWiki:Autosumm-blank", 0, ref autosummBlank, false);
+            Messages.Add("Blocklogentry", new MessagesOption(3, "blockRegex", true));
+            Messages.Add("Unblocklogentry", new MessagesOption(0, "unblockRegex", false));
+            Messages.Add("Reblock-logentry", new MessagesOption(3, "reblockRegex", false));
+            Messages.Add("Autosumm-blank", new MessagesOption(0, "autosummBlank", false));
             // autosummReplace is nonStrict because some large wikis don't include the "profanity" in their
             // messages (privacy measure?)
-            GenerateRegex("MediaWiki:Autosumm-replace", 1, ref autosummReplace, true);
+            Messages.Add("Autosumm-replace", new MessagesOption(1, "autosummReplace", true));
+
+            GetInterfaceMessages(Messages);
 
             GenerateRegexen();
         }
 
-        void GenerateRegex(string mwMessageTitle, int reqCount, ref string destRegex, bool nonStrict)
+        void GetInterfaceMessages(Dictionary<string, MessagesOption> Messages)
         {
-            // Get raw wikitext
-            string mwMessage = CVNBotUtils.GetRawDocument(rooturl + "w/index.php?title=" + mwMessageTitle + "&action=raw&usemsgcache=yes");
+            logger.InfoFormat("Fetching InterfaceMessages from {0}", rooturl);
+
+            string CombinedMessages = string.Join("|", Messages.Keys);
 
+            string sMwMessages = CVNBotUtils.GetRawDocument(
```

Comment:
What does "s" stand for? Might be better to use a  longer variable name, or to omit the s and explain its aspect with a code comment instead.